Enable Error on missing refs, Cleanup makefiles

LeroyR opened the pull request:
There are missing refs in the released rulebook:
![image](https://github.com/RoboCupAtHome/RuleBook/assets/14273072/9d4f18fe-f1b6-4204-8e43-0ef1da19bfe5)

This PR does not fix that.


But as i tried to reproduce the CI document with (revision\n_0 in footer) 
![image](https://github.com/RoboCupAtHome/RuleBook/assets/14273072/5d57ef45-f08a-4476-b0e8-b18302b033df)

I started by updating the build process to `latexmk`. 

- moved all "documents" into documents
  - let the makefile generate a rule for each tex file there
- removed unused stuff from the makefile
- add lint target to run chktex (as lint ci stage does)
- use stable ci image (texlive/texlive:TL2022-historic)
- update latexdockercmd.sh to just run "make all"

This changes nothing, it is simply a CI fault (checkout action discards .git maybe)?

But enabling -Werror and having less output reveals that we have missing refs:

```
Latexmk: ====List of undefined refs and citations:
  Reference `award:hri' on page 10 undefined on input line 114
  Reference `rule:not_attending' on page 14 undefined on input line 32
  Reference `rule:start_signal' on page 14 undefined on input line 32
  Reference `rule:not_attending' on page 17 undefined on input line 14
  Reference `rule:start_signal' on page 17 undefined on input line 14
  Reference `rule:perceived_intelligence' on page 20 undefined on input line 29
  Reference `rule:not_attending' on page 20 undefined on input line 29
 And 13 more --- see log file.

```



These are defined in `general_rules/PenaltiesBonuses.tex` which is not included in the rulebook.
e.g. in `rulebook_pages/GeneralRules.tex` we have the line `%\input{general_rules/PenaltiesBonuses}` commented.

We can merge this after fixing the Rulebook to catch these errors in CI in the future.

MatthijsBurgh commented:
When you are busy, please include the roadmap document as well. That one was missed in https://github.com/RoboCupAtHome/RuleBook/pull/831#issuecomment-1939094682

LeroyR commented:
Sorry, this is a big one.


### Cleanup build
- build changed to using latexmk with -Werror on  texlife/texlife2022 image, to catch missing refs during ci
  - you can test the ci build locally with `latexdockercmd.sh`
  
### fix missing refs in the rulebook
- added bonus/penalties section to rulebook (for social survery)
- removed scoresheet refs to `start_signal` for now as we would need to add the section which refers `inspection` which refers `robot` - dont know what to move where so i removed the ref from the scoresheet for now

### Add missing roadmap document
